supabase/functions/mailgun/index.ts

```diff
@@ -0,0 +1,184 @@
+    const from = data.get('from');
+    if (!from) return new Response('Missing parameter: from', { status: 406 });
+    const subject = data.get('subject');
+    if (!subject)
+        return new Response('Missing parameter: subject', { status: 406 });
+    const strippedText = data.get('stripped-text');
+    if (!strippedText)
+        return new Response('Missing parameter: stripped-text', {
+            status: 406,
+        });
+
+    const noteContent = getNoteContent(
+        subject.toString(),
+        strippedText.toString()
+    );
+
+    const { email: salesEmail } = extractMailContactData(from.toString());
+    if (!salesEmail)
+        return new Response(
+            `Could not extract sales email from From: ${from}`,
+            { status: 406 }
+        );
+    const { data: sales, error: fetchSalesError } = await supabaseAdmin
+        .from('sales')
+        .select('*')
+        .eq('email', salesEmail)
+        .maybeSingle();
+    if (fetchSalesError)
+        return new Response(
+            `Could not fetch sales from database, email: ${salesEmail}`,
+            { status: 500 }
+        );
+    if (!sales)
+        return new Response(
+            `Unable to find sales in database, email: ${salesEmail}`,
+            { status: 406 }
+        );
+
+    const { firstName, lastName, email, domain } = extractMailContactData(
+        to.toString()
+    );
+    if (!email)
+        return new Response(`Could not extract email from To: ${to}`, {
+            status: 406,
+        });
+
+    const { data: existingContact, error: fetchContactError } =
+        await supabaseAdmin
+            .from('contacts')
+            .select('*')
+            .eq('email', email)
+            .maybeSingle();
+    if (fetchContactError)
+        return new Response(
+            `Could not fetch contact from database, email: ${email}`,
+            { status: 500 }
+        );
+
+    // deno-lint-ignore no-explicit-any
+    let contact: any = undefined;
+    if (existingContact) {
+        contact = existingContact;
+    } else {
+        // If the contact does not exist, we need to create it, along with the company if needed
+        const { data: existingCompany, error: fetchCompanyError } =
+            await supabaseAdmin
+                .from('companies')
+                .select('*')
+                .eq('name', domain)
+                .maybeSingle();
+        if (fetchCompanyError)
+            return new Response(
+                `Could not fetch companies from database, name: ${domain}`,
+                { status: 500 }
+            );
+
+        // deno-lint-ignore no-explicit-any
+        let company: any = undefined;
+        if (existingCompany) {
+            company = existingCompany;
+        } else {
+            const { data: newCompanies, error: createCompanyError } =
+                await supabaseAdmin
+                    .from('companies')
+                    .insert({ name: domain, sales_id: sales.id })
```

Comment:
I'd UcFirst the domain

Reply:
You're right, we should get rid of the TLD. But nothing more. 

Also, we may need to blacklist generic email domains like gmail, hotmail, etc, and attach the contact top no company in this case (if it's possible), or to an "indeterminate" company (if it's not).

Reply:
I'll do it in a separate PR